### DISCUSSION TITLE
麻雀のルールの確認

### DISCUSSION THREAD

**Discussion by takejohn** [misskey-dev/misskey#14351] [General]
mahjongブランチ(#13112)で作業をするにあたって、想定されている麻雀のルールを確認したい。
- 喰いタン: **あり**
- 後付け: **あり**
- 飛び: あり / なし
  - ありの場合、0点となった時点で飛び / 続行
- 親流れの条件: 連荘なし / 親の聴牌で連荘 / 親の和了のみ連荘
- 飜縛り: 常に1飜縛り / 5本場以上などで2飜縛り
- 流し満貫: **あり**
  - 流し満貫の成立時は**流局とする**
    - 流局とする場合、不聴罰符の支払い: 行う / 行わない
- 役満 (大三元・大四喜・四槓子) の包: あり / なし
  - 四槓子の包: あり / なし
- 国士無双について暗槓に対する搶槓: あり / なし
- 槓ドラをめくるタイミング: 暗槓即乗り・明槓後めくり / すべて即乗り / すべて後めくり
- 裏ドラ: **あり**
- 形式聴牌: あり / なし
- フリテンリーチ: あり / なし
- 符の導入
  - 切り上げ満貫: 採用 / 不採用
  - 連風牌の雀頭に加算する符: 2符 / 4符
- 途中流局: ありの場合、成立時に連荘するか
  - 九種九牌 ([packages/frontend/assets/mahjong/99.png](https://github.com/misskey-dev/misskey/blob/mahjong/packages/frontend/assets/mahjong/99.png)って画像があるから採用なんだと思う)
  - 四家立直
  - 四槓散了
  - 四風連打
  - 三家和
- ダブロン・トリロン / 頭ハネ
  - ダブロン・トリロンで親が和了った場合連荘するか
  - 積み棒: 上家取り / 各和了者に支払い
- 南入・西入・和了り止め・聴牌止め・ウマ・オカ: (東風戦・半荘戦の概念と終局を実装してからだからまだ先になりそう)

**Comment by syuilo** [comment]
確定しているところで言うと

- 喰いタン: あり
- 後付け: あり
- 裏ドラ: あり
- 流し満貫: あり (実装が面倒であれば後回しで良さそう)
  - 流し満貫の成立時は流局とする
